compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/FirDelegatedPropertyChecker.kt

```diff
@@ -47,8 +49,13 @@ object FirDelegatedPropertyChecker : FirPropertyChecker() {
             override fun visitElement(element: FirElement) = element.acceptChildren(this)
 
             override fun visitFunctionCall(functionCall: FirFunctionCall) {
-                val errorNamedReference = functionCall.calleeReference as? FirErrorNamedReference ?: return
-                if (errorNamedReference.source?.kind != FirFakeSourceElementKind.DelegatedPropertyAccessor) return
+                val hasReferenceError = checkFunctionReferenceErrors(functionCall)
+                if (isGet && !hasReferenceError) checkReturnType(functionCall)
```

Comment:
I expected `DELEGATE_SPECIAL_FUNCTION_NONE_APPLICABLE` case will make `hasReferenceError` `true`, but how can both diagnostics be reported?

Reply:
One is reported on setValue and on is reported on getValue.

Reply:
Aha!